File: Java/printListReversely.java
/**
* 输入一个链表，从尾到头打印链表每个节点的值。
*/

/**
*    public class ListNode {
*        int val;
*        ListNode next = null;
*
*        ListNode(int val) {
*            this.val = val;
*        }
*    }
*
*/
import java.util.ArrayList;
public class Solution {
    public ArrayList<Integer> printListFromTailToHead(ListNode listNode) {
        ArrayList<Integer> a = new ArrayList<Integer> ();
        while (listNode != null) {
            a.add(0, listNode.val);
            listNode = listNode.next;
        }
        return a;
    }
}

File: Python/printListReversely.py
# -*- coding:utf-8 -*-

# 输入一个链表，从尾到头打印链表每个节点的值。

# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # 返回从尾部到头部的列表值序列，例如[1,2,3]
    def printListFromTailToHead(self, listNode):
        # write code here
        a = []
        while listNode:
            a.insert(0, listNode.val)
            listNode = listNode.next
        return a